databricks/koalas/series.py

```diff
@@ -4324,6 +4325,149 @@ def pct_change(self, periods=1):
+                3    6.0
+                4    6.0
+                Name: 0, dtype: float64
+
+                whereas pandas looks like the below.
+
+                >>> pser1 = kser1.to_pandas()
+                >>> pser2 = kser2.to_pandas()
+                >>> pser1.combine(pser2, lambda x, y: x + y).sort_index()
+                0    6
+                1    6
+                2    6
+                3    6
+                4    6
+                Name: 0, dtype: int64
+                >>> reset_option("compute.ops_on_diff_frames")
+
+        Parameters
+        ----------
+        other : Series or scalar
+            The value(s) to be combined with the `Series`.
+        func : function
+            Function that takes two scalars as inputs and returns an element.
+        fill_value : scalar, optional
+            The value to assume when an index is missing from
+            one Series or the other. The default specifies to use the
+            appropriate NaN value for the underlying dtype of the Series.
+
+        Returns
+        -------
+        Series
+            The result of combining the Series with the other object.
+
+        See Also
+        --------
+        Series.combine_first : Combine Series values, choosing the calling
+            Series' values first.
+
+        Examples
+        --------
+        Consider 2 Datasets ``s1`` and ``s2`` containing
+        highest clocked speeds of different birds.
+
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s1 = ks.Series({'falcon': 330.0, 'eagle': 160.0})
+        >>> s1.sort_index()
+        eagle     160.0
+        falcon    330.0
+        Name: 0, dtype: float64
+        >>> s2 = ks.Series({'falcon': 345.0, 'eagle': 200.0, 'duck': 30.0})
+        >>> s2.sort_index()
+        duck       30.0
+        eagle     200.0
+        falcon    345.0
+        Name: 0, dtype: float64
+
+        Now, to combine the two datasets and view the highest speeds
+        of the birds across the two datasets
+
+        >>> s1.combine(s2, max).sort_index()
+        duck        NaN
+        eagle     200.0
+        falcon    345.0
+        Name: 0, dtype: float64
+
+        In the previous example, the resulting value for duck is missing,
+        because the maximum of a NaN and a float is a NaN.
+        So, in the example, we set ``fill_value=0``,
+        so the maximum value returned will be the value from some dataset.
+
+        >>> s1.combine(s2, max, fill_value=0).sort_index()
+        duck       30.0
+        eagle     200.0
+        falcon    345.0
+        Name: 0, dtype: float64
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        # TODO: only `func` with return numeric-type is supported for now,
+        # assuming this case is the most frequent.
+        # because we can't infer the return type of `func` before it actually return.
+        @udf("double")
```

Comment:
We should not use the regular `udf`. This is super slow (can be 100+ times slower then other expressions).